spring-kafka/src/main/java/org/springframework/kafka/config/RetryTopicConfigurationSupport.java

```diff
@@ -0,0 +1,486 @@
+		 * @param exceptions the exceptions.
+		 * @return the configurer.
+		 * @see DefaultErrorHandler
+		 */
+		@SuppressWarnings("varargs")
+		@SafeVarargs
+		public final BlockingRetriesConfigurer retryOn(Class<? extends Exception>... exceptions) {
+			this.retryableExceptions = exceptions;
+			return this;
+		}
+
+		/**
+		 * Set the {@link BackOff} that should be used with the blocking retry mechanism.
+		 * By default, a {@link FixedBackOff} with 0 delay and 9 retry attempts
+		 * is configured. Note that this only has any effect for exceptions specified
+		 * with the {@link #retryOn} method - by default blocking retries are disabled
+		 * when using the non-blocking retries feature.
+		 * @param backoff the {@link BackOff} instance.
+		 * @return the configurer.
+		 * @see DefaultErrorHandler
+		 */
+		public BlockingRetriesConfigurer backOff(BackOff backoff) {
+			this.backOff = backoff;
+			return this;
+		}
+	}
+
+	/**
+	 * Configure the {@link KafkaConsumerBackoffManager} instance.
+	 */
+	public static class KafkaBackOffManagerConfigurer {
+
+		boolean timingAdjustmentEnabled = true;
+
+		private Integer maxThreadPoolSize = null;
+
+		private Clock clock;
+
+		/**
+		 * Disable timing adjustment for the delays. By choosing this option records
+		 * won't be processed exactly at the proper time. It's guaranteed however that
+		 * records won't be processed before their due time.
+		 * @return the configurer.
+		 * @see WakingKafkaConsumerTimingAdjuster
+		 */
+		public KafkaBackOffManagerConfigurer disableTimingAdjustment() {
+			this.timingAdjustmentEnabled = false;
+			return this;
+		}
+
+		/**
+		 * Set the maximum thread pool size to be used by the
+		 * {@link WakingKafkaConsumerTimingAdjuster}. This
+		 * {@link KafkaConsumerTimingAdjuster} implementation spawns threads that will
+		 * sleep for a calculated time, and after that will
+		 * {@link org.apache.kafka.clients.consumer.Consumer#wakeup()} the consumer, in
+		 * order to improve delay precision.
+		 * @param maxThreadPoolSize the maximum thread pool size.
+		 * @return the configurer.
+		 */
+		public KafkaBackOffManagerConfigurer setMaxThreadPoolSize(int maxThreadPoolSize) {
+			this.maxThreadPoolSize = maxThreadPoolSize;
+			return this;
+		}
+
+		/**
+		 * Set the {@link Clock} instance to be used with the
+		 * {@link KafkaConsumerBackoffManager}.
+		 * @param clock the clock instance.
+		 * @return the configurer.
+		 */
+		public KafkaBackOffManagerConfigurer setClock(Clock clock) {
+			this.clock = clock;
+			return this;
+		}
+	}
+
+	/**
+	 * Configure customizers for components instantiated by the retry topics feature.
+	 */
+	public static class CustomizersConfigurer {
+
+		private Consumer<CommonErrorHandler> errorHandlerCustomizer;
+		private Consumer<ConcurrentMessageListenerContainer<?, ?>> listenerContainerCustomizer;
+		private Consumer<DeadLetterPublishingRecoverer> deadLetterPublishingRecovererCustomizer;
```

Comment:
Every member of the class has to be surrounded with blank lines (including the last one in the class).